apps/api/src/domain/models/audio-item/entities/audio-item.entity.ts

```diff
@@ -90,7 +92,8 @@ export class AudioItem<T extends CoscradText = string> extends Resource {
 
         this.lengthMilliseconds = lengthMilliseconds;
 
-        this.transcript = !isNullOrUndefined(transcript) ? new Transcript(transcript) : null;
+        // It's a bit odd that we allow the invalid value to be set so that our validator catches this
+        this.transcript = !isNullOrUndefined(transcript) ? new Transcript(transcript) : transcript;
```

Comment:
would we do this higher up in the flow?

Reply:
We had a chicken-and-egg situation. We have a DTO from the database. We only want to create an instance if the DTO is valid. But we need an instance to polymorphically call invariant validation. so we have to build an invalid instance in the factory then call `validateInvariants` then decide whether to return errors or the built instance. It's not ideal, but the simplicity \ readability upstream was worth the trade-off when we designed this.